litestar/dto/_backend.py

```diff
@@ -365,15 +396,24 @@ def _create_transfer_type(
         field_definition: FieldDefinition,
```

Comment:
@peterschutt 

There's quite a few of those attributes that we keep passing around. Do you think it might be worth introducing some sort of context object here or refactoring this to some kind of builder pattern where these are instance attributes?

Reply:
I have no doubt there are refactorings that can be made through these parts and I could see how a context object could simplify these signatures and perhaps could also house the logic that modifies the `exclude` and `include` for the level of nesting.

From taking a quick look I can see other things I'd like to tidy up.

Raising `RecursionError` to break out for max nested depth is an interesting choice. Catching a `RecursionError` and continuing makes my spidey sense tingle:

https://github.com/litestar-org/litestar/blob/7414f7fd7d4782223502895e6a23b77ed635cd2d/litestar/dto/_backend.py#L130-L131

And this:

https://github.com/litestar-org/litestar/blob/7414f7fd7d4782223502895e6a23b77ed635cd2d/litestar/dto/_backend.py#L384-L386

Note we call `parse_model()` with `field_definition.annotation`, and then in 2nd line of that method we go right on ahead and create a new one:

https://github.com/litestar-org/litestar/blob/7414f7fd7d4782223502895e6a23b77ed635cd2d/litestar/dto/_backend.py#L110

On that last one, it should be a rule that we never unwrap the annotations from their parsed type object until we absolutely have to.